### DISCUSSION TITLE
Script editor config

### DISCUSSION THREAD

**Discussion by positlabs** [cables-gl/cables_docs#204] [Answers & Questions]
I found this empty section in the docs: https://cables.gl/docs/faq/ui_editor/ui_editor

I was hoping to find some way to configure the script editor and just get better at using it in general. I would like to: 
- Set tabs vs spaces
- Use an auto format command
- Maybe define an editor config
- Typescript support?

**Comment by steam0r** [comment]
hey,

we are using "ace" https://ace.c9.io/ and are quire happy with the current setting. formating can be done in userops by pressing this button on the top. this is not available for CustomOp, though.

![Bildschirmfoto 2022-03-17 um 20 48 52](https://user-images.githubusercontent.com/151271/158884023-7620e72e-0c6e-4c7c-9ff2-3b188b0c9464.png)

once you pressed that the code will be formated to our styleguide and rules, so no way to make your own rules, sorry :)

as for typescript: that would need a transpiler on the serverside, right? we have no plans on supporting typescript. 
any reason why you need this?

**Reply by positlabs** [reply]
In my desktop environment, I'm accustomed to using a key shortcut to format my code. It would be nice to have something like that for scripting in cables too. 

It's not a big deal if I have to follow a particular style guide. I mainly like formatting for consistency and uh... laziness. 

I don't necessarily need typescript, but sometimes I find code examples that are written in TS and it's just a pain to have to convert. I'm generally fine with vanilla JS but I do like the improved code hints that come with typescript. I haven't tried it before, but it looks like there are some browser-side compilers for typescript. Looks like ace supports typescript too. 

https://github.com/basarat/typescript-script
https://github.com/ajaxorg/ace/search?q=typescript

I guess one more question I had was about opening the code editor back up. I find myself hitting escape or clicking on the wrong thing and the editor closes. Is there a key shortcut to open it back up?

**Comment by pandrr** [comment]
i guess it would be nice to have some of the ace options in user prefs

some of those: https://github.com/ajaxorg/ace/wiki/Configuring-Ace

also theme selection would be nice
